src/main/webapp/index.xhtml

```diff
@@ -36,25 +36,32 @@
                 <h:message id="trade-err" class="error" for="trade-btn"/>
                 <h:message id="errmsg" class="error" for="bcode"/>
                 <h:outputText id ="totalPrice" value = "#{localTradeList.totalValue}"/>
-            </h:form>
             <!-- dataTable instead of ui:repeat for clear and easy table -->
-            <h:dataTable id="trade-list" 
-                         value="#{localTradeList.items}" 
-                         var="item"
-                         styleClass="trade-table hidden"
-                         style="display:#{localTradeList.items.isEmpty() ? 'none;' : 'block'}">
-                <h:column>
-                    <f:facet name="header">Barcode</f:facet>
-                    #{item.barcode}
-                </h:column>
-                <h:column>
-                    <f:facet name="header">Item</f:facet>
-                    #{item.name}
-                </h:column>
-                <h:column>
-                    <i class="far fa-trash-alt"></i>
-                </h:column>
-            </h:dataTable>
+            	<h:dataTable id="trade-list" 
+                         	value="#{localTradeList.items}" 
+                         	var="item"
+                         	styleClass="trade-table hidden"
+                         	style="display:#{localTradeList.items.isEmpty() ? 'none;' : 'block'}">
+               	 <h:column>
+                    	<f:facet name="header">Barcode</f:facet>
+                    	#{item.barcode}
+                	</h:column>
+                	<h:column>
+                    	<f:facet name="header">Item</f:facet>
+                    	#{item.name}
+                	</h:column>
+                	<h:column>
+                		<h:commandButton
+                			id="deleteItem"
+                			value="delete"
+                			icon = "fa-trash-alt"
+                			action="#{localTradeList.deleteItem(item)}"
+                			>
+                			<f:ajax execute="@form" resetValues="true" render="@form trade-list" />
+                    	</h:commandButton>

```

Comment:
commandButton resolves to an input element of type 'submit' so an inner element will never be valid and this is why the icon won't work; commandLink however wraps it in an anchor element which can have inner element allowing for the icon but you will have to change the CSS to avoid the default blue color.

Reply:
made the suggested change